traits/trait_set_object.py

```diff
@@ -263,35 +441,252 @@ def __reduce_ex__(self, protocol=None):
             self.__getstate__(),
         )
 
-    def __getstate__(self):
-        result = self.__dict__.copy()
-        result.pop("object", None)
-        result.pop("trait", None)
-        return result
-
-    def __setstate__(self, state):
-        name = state.setdefault("name", "")
-        object = state.pop("object", None)
-        if object is not None:
-            self.object = ref(object)
-            self.rename(name)
-        else:
-            self.object = lambda: None
-
-        self.__dict__.update(state)
-
     def __ior__(self, value):
+        """ Return self|=value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                An empty set.
+            added : set
+                Set of added values.
+
+        """
         self.update(value)
         return self
 
     def __iand__(self, value):
+        """  Return self&=value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : set
+                An empty set.
+
+
+        """
         self.intersection_update(value)
         return self
 
     def __ixor__(self, value):
+        """ Return self ^= value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : set
+                A set containing the added values.
+
+        """
         self.symmetric_difference_update(value)
         return self
 
     def __isub__(self, value):
+        """ Return self-=value.
+
+        Parameters
+        ----------
+        value : any
+            A value.
+
+        Returns
+        -------
+        self : set
+            The updated set.
+
+        Notes
+        -----
+        Parameters in the notification:
+            removed : set
+                A set containing the removed values.
+            added : Undefined
+                Will be Undefined.
+
+        """
         self.difference_update(value)
         return self
+
+
+class TraitSetObject(TraitSet):
+    """ A specialization of TraitList with a default validator and notifier
+    which provide bug-for-bug compatibility with the TraitsListObject from
```

Comment:
Ooops, missed this. All mention of `TraitList` should be changed to `TraitSet`, and `TraitsListObject` to `TraitSetObject`. And there are no bug-for-bug implementations here is there?

Reply:
And there are a lot of mentions of "list" in this docstring, they should all be changed to "set".

Reply:
:man_facepalming: Sorry, changed it.